agent/src/main/java/com/testsigma/agent/mobile/ios/IosDeviceService.java

```diff
@@ -131,7 +173,28 @@ public void cleanupWda(MobileDevice device) {
     }
   }
 
-  public String installApp(MobileDevice device, String appUrl) throws AutomatorException {
-    return new AppInstaller(httpClient).installApp(device.getName(), device.getUniqueId(), appUrl);
+  public String installApp(MobileDevice device, String appUrl, Boolean isEmulator) throws AutomatorException {
+    return new AppInstaller(httpClient).installApp(device.getName(), device.getUniqueId(), appUrl, isEmulator);
   }
+
+  public MobileDevice getSimulatorDevice(String udid) throws AutomatorException, TestsigmaException {
+    IosDeviceCommandExecutor iosDeviceCommandExecutor = new IosDeviceCommandExecutor();
+    Process p = iosDeviceCommandExecutor.runDeviceCommand(new String[]{"describe", "--udid", udid, "--json"}, false);
+    String deviceDescriptionJson = iosDeviceCommandExecutor.getProcessStreamResponse(p);
+    JSONObject device = getSimulatorProperties(deviceDescriptionJson);
+    MobileDevice mobileDevice = new MobileDevice();
+    mobileDevice.setOsName(MobileOs.IOS);
+    mobileDevice.setUniqueId(device.getString("udid"));
+    mobileDevice.setName(device.getString("name"));
+    mobileDevice.setOsVersion(device.getString("os_version").split(" ")[1]);
+    mobileDevice.setApiLevel(mobileDevice.getOsVersion());
+    mobileDevice.setAbi(device.getString("architecture"));
+    mobileDevice.setIsOnline(device.getString("state").equals("Booted"));
```

Comment:
We need to handle Null pointers in these. Earlier we have seen that some devices does not give all the properties we are expecting

Reply:
How do you suggest we handle null values ? There are some values which are absolutely essential. For example, uniqueId, OsVersion, Name. For those we can throw an exception if null. For others, even if there are null values, executions might be successful. For those handling null values might not be required. 

Note that only when the idb command produces an incorrect output, this problem might arise.